libs/cluster/Session/MigrateCommand.cs

```diff
@@ -108,59 +112,48 @@ private bool TryMIGRATE(int count, byte* ptr)
+                if (option.EqualsUpperCaseSpanIgnoringCase("COPY"u8))
                     copyOption = true;
-                else if (option.Equals("REPLACE", StringComparison.OrdinalIgnoreCase))
+                else if (option.EqualsUpperCaseSpanIgnoringCase("REPLACE"u8))
                     replaceOption = true;
-                else if (option.Equals("AUTH", StringComparison.OrdinalIgnoreCase))
+                else if (option.EqualsUpperCaseSpanIgnoringCase("AUTH"u8))
                 {
-                    if (!RespReadUtils.ReadStringWithLengthHeader(out passwd, ref ptr, recvBufferPtr + bytesRead))
-                        return false;
-                    args--;
+                    passwd = parseState.GetString(currTokenIdx++);
                 }
-                else if (option.Equals("AUTH2", StringComparison.OrdinalIgnoreCase))
+                else if (option.EqualsUpperCaseSpanIgnoringCase("AUTH2"u8))
                 {
-                    if (!RespReadUtils.ReadStringWithLengthHeader(out username, ref ptr, recvBufferPtr + bytesRead))
-                        return false;
-                    if (!RespReadUtils.ReadStringWithLengthHeader(out passwd, ref ptr, recvBufferPtr + bytesRead))
-                        return false;
-                    args -= 2;
+                    username = parseState.GetString(currTokenIdx++);
+                    passwd = parseState.GetString(currTokenIdx++);
                 }
-                else if (option.Equals("KEYS", StringComparison.OrdinalIgnoreCase))
+                else if (option.EqualsUpperCaseSpanIgnoringCase("KEYS"u8))
                 {
                     slots = [];
                     if (transferOption == TransferOption.SLOTS)
                         pstate = MigrateCmdParseState.MULTI_TRANSFER_OPTION;
 
                     transferOption = TransferOption.KEYS;
                     keys = new Dictionary<ArgSlice, KeyMigrationStatus>(ArgSliceComparer.Instance);
-                    while (args > 0)
+                    while (currTokenIdx < parseState.Count)
                     {
-                        byte* keyPtr = null;
-                        var ksize = 0;
-
-                        if (!RespReadUtils.ReadPtrWithLengthHeader(ref keyPtr, ref ksize, ref ptr, recvBufferPtr + bytesRead))
-                            return false;
-                        args--;
+                        var currKeySlice = parseState.GetArgSliceByRef(currTokenIdx++);
+                        var sbKey = currKeySlice.SpanByte;
 
                         // Skip if previous error encountered
                         if (pstate != MigrateCmdParseState.SUCCESS) continue;
 
                         // Check if all keys are local R/W because we migrate keys and need to be able to delete them
-                        var slot = HashSlotUtils.HashSlot(keyPtr, ksize);
+                        var slot = HashSlotUtils.HashSlot(sbKey.ToPointer(), sbKey.Length);
```

Comment:
Maybe not for this PR, but ArgSlice might need access method to the ptr so we won't need to have a SpanByte variable here.